plugins/TagFix_Housenumber.py

```diff
@@ -26,20 +26,32 @@ class TagFix_Housenumber(Plugin):
 
     def init(self, logger):
         Plugin.init(self, logger)
-        self.errors[10] = { "item": 2060, "level": 3, "tag": ["addr", "fix:survey"], "desc": T_(u"addr:housenumber does not start by a number") }
-        self.errors[14] = { "item": 2060, "level": 3, "tag": ["addr", "fix:chair"], "desc": T_(u"Invalid tag on interpolation way") }
-        self.errors[15] = { "item": 2060, "level": 3, "tag": ["addr", "fix:chair"], "desc": T_(u"Invalid addr:interpolation or addr:inclusion value") }
-        self.CountryCZ = self.father.config.options.get("country") == "CZ"
-        self.CountryHousenumberWithoutNumber = self.father.config.options.get("country") in ('RU', 'BG')
+        self.errors[10] = {"item": 2060, "level": 3, "tag": ["addr", "fix:survey"], "desc": T_(u"addr:housenumber does not start by a number")}
+        self.errors[14] = {"item": 2060, "level": 3, "tag": ["addr", "fix:chair"], "desc": T_(u"Invalid tag on interpolation way")}
+        self.errors[15] = {"item": 2060, "level": 3, "tag": ["addr", "fix:chair"], "desc": T_(u"Invalid addr:interpolation or addr:inclusion value")}
+        self.Country = self.father.config.options.get("country")
+        import re
+        from collections import defaultdict
+
+        # By default, validate that house number starts with 1-9
+        housenumberRegexByCountry = defaultdict(re.compile("^[1-9]"))
+
+        # More specific rules by country
+
+        # From open data from cantons Zurich and Bern. See also https://github.com/ltog/osmi-addresses/issues/93
+        housenumberRegexByCountry["CH"] = re.compile("^[1-9][0-9]{0,3}( ?[a-zA-Z])?$")
+
+        # https://wiki.openstreetmap.org/wiki/Cs:WikiProject_Czech_Republic/Address_system
+        housenumberRegexByCountry["CZ"] = re.compile("^(ev\.)?[1-9]")
+
+        # From open data from CACLR, https://data.public.lu/en/datasets/registre-national-des-localites-et-des-rues/
+        housenumberRegexByCountry["LU"] = re.compile("^[1-9][0-9]{0,3}([A-Z]){0,3}(-[1-9][0-9]{0,3}([A-Z]){0,3})?$")
 
     def node(self, data, tags):
         err = []
-        if self.CountryHousenumberWithoutNumber:
-          return []
-        if "addr:housenumber" in tags and (len(tags["addr:housenumber"]) == 0 or not (
-            tags["addr:housenumber"][0].isdigit() or
-            (self.CountryCZ and tags["addr:housenumber"].startswith('ev.') and tags["addr:housenumber"][3].isdigit())
-            )):
+        if self.Country in ('RU', 'BG'):  # Countries with no house numbers
+            return []
```

Comment:
For 'RU' and 'BG' we have now a now a `not_for`
https://github.com/osm-fr/osmose-backend/blob/master/plugins/Source.py#L28